static/js/publisher/release/components/modal.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { PropTypes } from "prop-types";
+
+import { closeModal } from "../actions/modal";
+
+import * as actions from "../actions/";
+
+class ModalActionButton extends Component {
+  constructor(props) {
+    super(props);
+
+    this.onClickHandler = this.onClickHandler.bind(this);
+    this.state = {
+      loading: false
+    };
+  }
+
+  onClickHandler() {
+    const { onClickAction, dispatch } = this.props;
+
+    if (onClickAction.reduxAction) {
+      const { reduxAction } = onClickAction;
+      if (
+        this.props[reduxAction] &&
+        typeof this.props[reduxAction] == "function"
+      ) {
+        // If an action is passed, perform the specific action
+        this.props[onClickAction.reduxAction]();
+      }
+    } else {
+      // Otherwise dispatch the action object
+      dispatch(onClickAction);
+    }
+
+    this.setState({
+      loading: true
+    });
+  }
+
+  render() {
+    const { appearance, children } = this.props;
+    const { loading } = this.state;
+
+    const className = [
+      `p-button--${appearance}`,
+      "u-no-margin--bottom",
+      "u-float--right",
+      ["positive", "negative"].indexOf(appearance) > -1 ? "is--dark" : ""
+    ];
+
+    return (
+      <button
+        className={className.join(" ")}
+        onClick={this.onClickHandler}
+        disabled={loading}
+      >
+        {loading && "Loading..."}
+        {!loading && children}
+      </button>
+    );
+  }
+}
+
+ModalActionButton.propTypes = {
+  onClickAction: PropTypes.object.isRequired,
+  appearance: PropTypes.string.isRequired,
+  children: PropTypes.node.isRequired,
+  dispatch: PropTypes.func.isRequired
+};
+
+const mapActionButtonDispatchToProps = dispatch => {
+  const dispatchObj = {
+    dispatch
+  };
+
+  Object.keys(actions).forEach(actionKey => {
+    if (typeof actions[actionKey] === "function") {
+      dispatchObj[actionKey] = () => dispatch(actions[actionKey]());
```

Comment:
While it's nice that in theory any action can be dispatched this way it quickly fails because most of actions creator functions take arguments which are not passed here in any way.
So this will only work for actions that don't take any arguments.
Because of that it may be better to explicitly list only actions that are supported.

Reply:
we could pass an `actionArguments` array? But we run the risk of those arguments having functions etc. passed through, then we need to check those etc. etc... I think a whitelist is a good idea that can be added to as needed

Reply:
We don't need a full blown solution. It's safe to assume that for now we can only trigger actions without arguments from modal. At least as functions. You can always pass an action object, which should cover most cases anyway.
So I would just explicitly list only `setDefaultTrack` and `clearDefaultTrack`.